Latest & greatest version of Stapler

kohsuke opened the pull request.

kohsuke committed:
Latest & greatest

kohsuke commented:
Why is it picking up bits from `http://repo.cloudbees.com/content/repositories/central`? It should be looking at `https://repo.jenkins-ci.org/public`

batmat commented:
Nice open/close christmas tree ;-).
NIT: maybe adding stapler in the commit title would improve a slightly little wee bit that PR.

kohsuke commented:
At least one of the failures look legit. I need to look at this.

kohsuke committed:
Merge remote-tracking branch 'origin/master' into stapler-1.240

kohsuke committed:
Fixing a test

Previously, when the 2nd `e.click()` is called, the page has already transitioned to another page, so the anchor objects have become invalid.

HtmlUnit somehow still manages to follow a link, except when there's some JavaScript attached to the onclick() handler it's going to execute in a strange environment. Some management link requires a POST, and that was done via JavaScript. So that broke the test.

kohsuke commented:
Free +1 opportunities for eveyrone,  @reviewbybees

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

andresrc commented:
:bee:

stephenc commented:
:bee:

jglick commented on test/src/test/java/hudson/model/ManagementLinkTest.java:
```diff
@@ -47,11 +48,15 @@
-        assertTrue(anchors.size()>=8);
-        for(HtmlAnchor e : (List<HtmlAnchor>) anchors) {
-            e.click();
+        WebClient wc = j.createWebClient();
```
How does this relate to a new version of Stapler?
